src/main/java/org/cryptomator/cryptofs/health/dirid/OrphanDir.java

```diff
@@ -134,10 +158,53 @@ Path prepareRecoveryDir(Path pathToVault, FileNameCryptor cryptor) throws IOExce
 		return new CryptoPathMapper.CiphertextDirectory(stepParentUUID, stepParentDir);
 	}
 
+	//visible for testing
+	Optional<String> retrieveDirId(Path orphanedDir, Cryptor cryptor) {
+		var dirIdFile = orphanedDir.resolve(Constants.DIR_ID_FILE);
+		var dirIdBuffer = ByteBuffer.allocate(36); //a dir id contains at most 36 ascii chars, in this impl encoded in utf8
+
+		try (var channel = Files.newByteChannel(dirIdFile, StandardOpenOption.READ); //
+			 var decryptingChannel = wrapDecryptionAround(channel, cryptor)) {
+			decryptingChannel.read(dirIdBuffer);
+			dirIdBuffer.flip();
+		} catch (IOException e) {
+			LOG.info("Unable to read dirIdFile of {}.", orphanedDir, e);
+			return Optional.empty();
+		}
+
+		var allegedDirId = StandardCharsets.UTF_8.decode(dirIdBuffer).toString();
+
+		var dirIdHash = orphanedDir.getParent().getFileName().toString() + orphanedDir.getFileName().toString();
+		if (dirIdHash.equals(cryptor.fileNameCryptor().hashDirectoryId(allegedDirId))) {
+			return Optional.of(allegedDirId);
+		} else {
+			LOG.info("Hash of read directory id {} does not match actual cipher dir hash {}.", allegedDirId, dirIdHash);
+			return Optional.empty();
+		}
+	}
+
+	//exists and visible for testability
+	DecryptingReadableByteChannel wrapDecryptionAround(ByteChannel channel, Cryptor cryptor) {
```

Comment:
I don't want to suppress any creativity, but standard names help understand method without looking at their implementation

```suggestion
	DecryptingReadableByteChannel createDecryptingReadableByteChannel(ByteChannel channel, Cryptor cryptor) {
```

Reply:
Fixed in [62dd511](https://github.com/cryptomator/cryptofs/pull/127/commits/62dd511a97ee2f73a3234367a3cb9a1cef7d106b)